Nodejs/Product/Nodejs/NodejsProject.cs

```diff
@@ -574,11 +572,16 @@ out frame
             return hr;
         }
 
+        private bool IsES6IntellisensePreview() {
+            // If the analysis level is set to preview then use the TypeScript language service for node.js
+            return NodejsPackage.Instance.IntellisenseOptionsPage.AnalysisLevel == Options.AnalysisLevel.Preview;
+        }
+
         #region IVsProject Members
 
         public int AddItem(uint itemidLoc, VSADDITEMOPERATION dwAddItemOperation, string pszItemName, uint cFilesToOpen, string[] rgpszFilesToOpen, IntPtr hwndDlgOwner, VSADDRESULT[] pResult) {
-            if (_innerProject3 != null && IsJavaScriptFile(pszItemName)) {
-                Guid ourEditor = typeof(NodejsEditorFactory).GUID;
+            if (_innerProject3 != null && IsJavaScriptFile(pszItemName)) {
+                Guid ourEditor = this.IsES6IntellisensePreview() ? Guid.Empty : typeof(NodejsEditorFactory).GUID;
```

Comment:
The TSLS editor registers the js file extension at a higher priority than the JSLS editor so it will be the first to process the request.